internal/pkg/service/buffer/api/http/http.go

```diff
@@ -78,11 +78,11 @@ func HandleHTTPServer(proc *servicectx.Process, d dependencies.ForServer, u *url
 		logger.Infof("HTTP %q mounted on %s %s", m.Method, m.Verb, m.Pattern)
 	}
 
-	proc.Add(func(ctx context.Context, errCh chan<- error) {
+	proc.Add(func(ctx context.Context, shutdown servicectx.ShutdownFn) {
```

Comment:
Na konci kazdeho testu sa vola cez `t.Cleanup` funkcia `servicectx.Process.Shutdown`.

V tomto PR potrebujem testovat ukoncenie procesu priamo v teste.

V tom pripade sa mi zavola duplicitne aj po skonceni testu, co nefungovalo.

Problem bol, ze tu je blokujuci `channel err`, preto som ho nahradil za callback `ShutodownFn`,
pomocou ktorého som implementoval, ze `Shutdown` sa moze zavolat viac krat, duplicitne volania sa ignoruju.

Reply:
Je to takto odolnejsie, ... teoreticky sa to moze stat aj v produkcii, ze nastanu 2 fatalne chyby "naraz".

Reply:
Je to takto odolnejsie, ... teoreticky sa to moze stat aj v produkcii, ze nastanu 2 fatalne chyby "naraz".

Reply:
Je to takto odolnejsie, ... teoreticky sa to moze stat aj v produkcii, ze nastanu 2 fatalne chyby "naraz".